orca-retrofit-reactive/src/main/kotlin/com/netflix/spinnaker/orca/reactive/WindowedRequestMetrics.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.orca.reactive
+
+import com.google.common.base.Ticker
+import com.google.common.cache.CacheBuilder
+import org.springframework.http.HttpStatus
+import java.time.Duration
+import java.util.concurrent.TimeUnit.MILLISECONDS
+
+/**
+ * A mechanism for recording request metrics that has a look-back limit. That is
+ * the metrics it reports represent those reported over the last [expireAfter]
+ * duration.
+ */
+class WindowedRequestMetrics
```

Comment:
This is the meat of the metrics implementation, giving us a rolling window of metrics from a given service or set of services.